catalogue.py

```diff
@@ -837,203 +902,226 @@ def cross_match(self,cat,radius='largest',join_type='1',redo=False,write=True):
                 self.set_key_fields()
             if len(cat.coords) == 0:
                 cat.set_key_fields()
 
-            #get sky coordinates and find nearest match of every source
-            #from this instance, independent of the search radius
+            # get sky coordinates and find nearest match of every source
+            # from this instance, independent of the search radius
             c1 = self.coords[self.name]
             c2 = cat.coords[cat.name]
-            indices,sep,sep3d = c1.match_to_catalog_sky(c2)
+            indices, sep, sep3d = c1.match_to_catalog_sky(c2)
 
-            #take the maximum radius from the two
+            # take the maximum radius from the two
             if radius == 'largest':
-                radius = max(self.search_rad,cat.search_rad)
+                radius = max(self.search_rad, cat.search_rad)
                 if self.verbose:
                     print('Using the largest of the two search radii of {0} arcsec.'.format(radius))
 
-            #only take matched rows from cat, so self.df and cat.df are parallel
+            # only take matched rows from cat, so self.df and cat.df are parallel
             cat.df = cat.df.iloc[indices].reset_index(drop=True)
 
-            #create pandas dataframe of the separations in arcsec
-            sep_col = '{0}_{1}_sep'.format(cat.name,self.name)
-            sepdf = pd.DataFrame(data = {sep_col : sep.arcsec})
+            # create pandas dataframe of the separations in arcsec
+            sep_col = '{0}_{1}_sep'.format(cat.name, self.name)
+            sepdf = pd.DataFrame(data={sep_col: sep.arcsec})
 
-            #only take matches within search radius
+            # only take matches within search radius
             indices = np.where(sep.arcsec < radius)[0]
 
-            #only add cross-matched table when at least 1 match
+            # only add cross-matched table when at least 1 match
             if len(indices) >= 1:
-                print("Found {0} matches within {1} arcsec.".format(len(indices),radius))
+                print("Found {0} matches within {1} arcsec.".format(len(indices), radius))
 
-                #don't reset indices so cat.df stays parallel with self.df
+                # don't reset indices so cat.df stays parallel with self.df
                 cat.df = cat.df.iloc[indices]
                 sepdf = sepdf.iloc[indices]
 
-                #concatenate tables together according to match type
+                # concatenate tables together according to match type
                 if join_type == '1and2':
-                    matched_df = pd.concat([self.df,cat.df,sepdf],axis=1,join='inner')
+                    matched_df = pd.concat([self.df, cat.df, sepdf], axis=1, join='inner')
                 elif join_type == '1':
-                    matched_df = pd.concat([self.df,cat.df,sepdf],axis=1,join_axes=[self.df.index])
+                    matched_df = pd.concat([self.df, cat.df, sepdf], axis=1,
```

Comment:
I had some problems with this line when I tried to run the code before. I think the `join_axes` argument is deprecated in newer versions of pandas. I had to change it to:
``` matched_df = pd.concat([self.df,cat.df,sepdf],axis=1).reindex(self.df.index)```

Reply:
I have fixed that.